doc/lag/Static Port Channel Design Document.md

```diff
@@ -0,0 +1,278 @@
+  * [4 Flow Diagrams](#4-flow-diagrams)
+    * [4.1 Port Channel Creation](#41-port-channel-creation)
+    * [4.2 Port Channel Member Addition](#42_port-channel-member-addition)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Serviceability and Debug](#6-serviceability-and-debug)
+  * [7 Warm Boot Support](#7-warm-boot-support)
+  * [8 Scalability](#8-scalability)
+  * [10 Unit Test](#9-unit-test)
+  * [11 References](#references)
+    * [Openconfig support for Port Channel](#openconfig-support-for-port-channel)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/31/2019  | Amitabha Sen       | Initial version                   |
+| 0.2 | 06/20/2022  | Kannan Selvaraj    | Updated load-balance round-robin  |
+|     |             |                    | show command display as STATIC    |
+|     |             |                    | CONFIG_DB flag is changed from    |
+|     |             |                    | static to on                      |
+
+# About this Manual
+This document provides details on achieving Static Port Channel functionality in SONiC. 
+
+# Scope
+This document describes the high level design of Static Port Channel functionality.
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| LAG                      | Link Aggregation Group              |
+| LACP                     | Link Aggregation Control Protocol   |
+
+# 1 Feature Overview
+## 1.1 Requirements
+### 1.1.1 Functional Requirements
+1. Support Port Channel in Static mode. Static Port Channel member ports are always available for traffic transmission and reception if the ports links are up. Port types and features are supported in the same way as SONiC LAGs today. MLAG should seamlessly work in Static Mode.
+
+### 1.1.2 Configuration and Management Requirements
+1. Provide configuration to set a Port Channel mode as on with the click CLI.
+2. Enhance show commands to display the Port Channel mode as STATIC when configured so with the click CLI.
+
+### 1.1.3 Scalability Requirements
+1. The maximum number of Port Channels configurable (Static and Dynamic put together) per system is 256.
+2. The maximum number of member ports configurable per Port Channel are 32. (Generally depends on ASIC)
+
+### 1.1.4 Warm Boot Requirements
+The traffic loss during a warm reboot should be within the prescribed limits of warm boot.
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+A Static Port Channel has its member ports always available for traffic as long the links are up. All ports are mandated to have the same speed to become members of a Port Channel. Teamd uses round-robin for a Static Port Channel.
```

Comment:
Please avoid using round-robin as discussed. If there is a bug in teamd we should raise an issue with teamd. Both LACP and static should use same hashing mechanism. If required we can expose hashing mechanism through CLI to the user

Reply:
Hi Kannan,

The commit https://github.com/jpirko/libteam/commit/deadb5b715227429a1879b187f5906b39151eca9 caused the issue of 100% cpu.

The revert commit https://github.com/jpirko/libteam/commit/61efd6de2fbb8ee077863ee5a355ac3dfd9365b9 has the details about the issue that you were mentioning in today's meeting.

You may check which libteam version is used in your testing and fix the above issue.

Reply:
Sudharsan,

Regarding exposing the hashing mechanism through CLI: the traffic hashing/distribution among the portchannel members can happen at 2 instances - (1) Kernel (2) Merchant Silicon.

Libteam gives provision to configure hashing at a kernel level.
But hashing/distribution at a silicon level is typically vendor specific. Achieving the same traffic distribution at kernel and silicon may not be possible as all the silicon may not have the exact match in the hashing algorithms.

Reply:
Yes will try loadbalance with fix, updated accordingly in the document